pkg/cloudprovider/providers/vsphere/nodemanager.go

```diff
@@ -121,16 +123,25 @@ func (nm *NodeManager) DiscoverNode(node *v1.Node) {
 				ctx, cancel := context.WithCancel(context.Background())
 				defer cancel()
 				vm, err := res.datacenter.GetVMByUUID(ctx, nodeUUID)
+				if err != nil {
+					glog.V(4).Infof("error while looking for vm=%+v in vc=%s and datacenter=%s",
+						node.Name, vm, res.vc, res.datacenter.Name())
+						continue
+				}
 				if vm != nil {
-					glog.V(4).Infof("Found node %s as vm=%+v in vc=%s and datacenter=%s",
+					glog.V(4).Infof("found node %s as vm=%+v in vc=%s and datacenter=%s",
 						node.Name, vm, res.vc, res.datacenter.Name())
 
 					nodeInfo := &NodeInfo{dataCenter: res.datacenter, vm: vm, vcServer: res.vc}
-					nm.SetNodeInfo(node.ObjectMeta.Name, nodeInfo)
+					nm.addNodeInfo(node.ObjectMeta.Name, nodeInfo)
 					for range queueChannel {}
+					mutex.Lock()
+					vmFound = true
+					mutex.Unlock()
 					break
+
 				} else {
-					glog.V(4).Infof("Did not find node %s in vc=%s and datacenter=%s",
+					glog.V(4).Infof("did not find node %s in vc=%s and datacenter=%s",
```

Comment:
Question - any particular reason why we dont start log messages with caps? I dont see any such convention followed. It looks odd to start a sentence with a lower case. If you agree, please fix this everywhere.

Reply:
This was recommended by kubernetes community